File: client-clojure/src/sirqul_io_t_platform/specs/notification_message_response.clj
(ns sirqul-io-t-platform.specs.notification-message-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.account-short-response :refer :all]
            )
  (:import (java.io File)))


(def notification-message-response-data
  {
   (ds/opt :sender) account-short-response-spec
   (ds/opt :event) string?
   (ds/opt :notificationMessage) string?
   (ds/opt :created) int?
   (ds/opt :hasRead) boolean?
   (ds/opt :contentId) int?
   (ds/opt :contentType) string?
   (ds/opt :contentName) string?
   (ds/opt :parentId) int?
   (ds/opt :parentType) string?
   (ds/opt :actionCategory) string?
   (ds/opt :thumbnailURL) string?
   (ds/opt :coverURL) string?
   })

(def notification-message-response-spec
  (ds/spec
    {:name ::notification-message-response
     :spec notification-message-response-data}))

File: client-scalaz/src/main/scala/org/openapitools/client/api/NotificationMessageResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import NotificationMessageResponse._

case class NotificationMessageResponse (
  sender: Option[AccountShortResponse],
event: Option[String],
notificationMessage: Option[String],
created: Option[Long],
hasRead: Option[Boolean],
contentId: Option[Long],
`contentType`: Option[String],
contentName: Option[String],
parentId: Option[Long],
parentType: Option[String],
actionCategory: Option[String],
thumbnailURL: Option[String],
coverURL: Option[String])

object NotificationMessageResponse {
  import DateTimeCodecs._

  implicit val NotificationMessageResponseCodecJson: CodecJson[NotificationMessageResponse] = CodecJson.derive[NotificationMessageResponse]
  implicit val NotificationMessageResponseDecoder: EntityDecoder[NotificationMessageResponse] = jsonOf[NotificationMessageResponse]
  implicit val NotificationMessageResponseEncoder: EntityEncoder[NotificationMessageResponse] = jsonEncoderOf[NotificationMessageResponse]
}
